--- /usr/local/opnsense/mvc/app/library/OPNsense/Auth/Local.php.orig	2025-07-01 08:08:34.895292000 +0000
+++ /usr/local/opnsense/mvc/app/library/OPNsense/Auth/Local.php	2025-07-01 08:56:50.594504000 +0000
@@ -166,7 +166,9 @@
             }
             if (password_verify($password, (string)$userObject->password)) {
                 // password ok, return successfully authentication
-                return true;
+		$_SESSION['mfa_pending'] = true;
+		$_SESSION['mfa_username'] = $username;
+		return true;
             }
         }
 
